app/src/main/java/a3/com/convo/activities/HomeScreenActivity.java

```diff
@@ -51,5 +58,109 @@ public void onClick(View view) {
+            Log.e("ProfileDetailsFragment", "User doesn't have name for some reason.");
+            return;
+        }
+
+        tvUserName.setText(userName);
+
+        String userHometownObjectId = user.getString(Constants.HOMETOWN);
+        if (userHometownObjectId == null) {
+            Log.e("ProfileDetailsFragment", "User doesn't have hometown and that's fine");
+            tvUserHometown.setText(Constants.EMPTY_STRING);
+        }
+        else {
+            ParseQuery<Page> query = ParseQuery.getQuery(Page.class);
+            query.getInBackground(userHometownObjectId, new GetCallback<Page>() {
+                @Override
+                public void done(Page object, ParseException e) {
+                    if (e == null && object != null) {
+                        String hometownName = object.getName();
+                        if (hometownName == null || hometownName.isEmpty()){
+                            Log.e("ProfileDetailsFragment", "User's hometown object has no name");
+                            return;
+                        }
+                        tvUserHometown.setText(getString(R.string.from) + " " + hometownName);
+                    }
+                    else {
+                        Log.e("hometown", "hometown not showing up");
+                        e.printStackTrace();
+                    }
+                }
+            });
+        }
+
+        Number gamesPlayedNum = user.getNumber(Constants.NUM_GAMES);
+        if (gamesPlayedNum == null || !(gamesPlayedNum instanceof Integer)) {
+            Log.e("ProfileDetailsFragment", "Num games played is null or not an integer.");
+            return;
+        }
+        Integer gamesPlayed = (Integer)gamesPlayedNum;
+        int userGamesPlayed = gamesPlayed.intValue();
+        tvNumGamesPlayed.setText(userGamesPlayed + " " + getString(R.string.num_games));
+
+        String profPicUrl = user.getString(Constants.PROF_PIC_URL);
+        if (profPicUrl != null) {
+            GlideApp.with(this)
+                    .load(profPicUrl)
+                    .circleCrop()
+                    .into(ivUserProfPic);
+        }
+        else {
+            Log.e("ProfileDetailsFragment", "User doesn't have profile picture and that's fine");
+        }
+
+        ArrayList<String> userLikes = (ArrayList<String>) user.get(Constants.PARSE_PAGE_LIKES_KEY);
+        if (userLikes == null || userLikes.isEmpty()) {
+            Log.e("ProfileDetailsFragment", "User has no likes.");
+            return;
+        }
+        else {
+            final String[] likesSentence = {"Some of the pages you like on facebook include...\n"};
+            for (int i = 0; i < 3 && i < userLikes.size(); i++) {
+                String likeObjectId = userLikes.get(i);
+                if (likeObjectId == null || likeObjectId.isEmpty()) {
+                    continue;
+                }
+                ParseQuery<Page> query = ParseQuery.getQuery(Page.class);
```

Comment:
Move this to a separate method

Reply:
Oops sorry I think I missed the file!

Reply:
I’ll just put all of the setting UI contents in separate functions so it’s cleaner.